lib/rule-helpers.js

```diff
@@ -0,0 +1,59 @@
+'use babel';
+
+const DOCUMENTATION_LIFETIME = 86400 * 1000; // 1 day TODO: Configurable?
+const docsRuleCache = new Map();
+
+export function takeWhile(source, predicate) {
+  const result = [];
+  const { length } = source;
+  let i = 0;
+
+  while (i < length && predicate(source[i], i)) {
+    result.push(source[i]);
+    i += 1;
+  }
+
+  return result;
+}
+
+// Retrieves style guide documentation with cached responses
+export async function getRuleMarkDown(rule) {
+  if (docsRuleCache.has(rule)) {
+    const cachedRule = docsRuleCache.get(rule);
+    if (cachedRule.expires >= new Date().getTime()) {
```

Comment:
Shouldn't this be `<=`?

Reply:
expires is a future date, so I think it is ok, but maybe I am worng.

```javascript
    docsRuleCache.set(rule, {
      markdown: markdownOutput,
      expires: new Date().getTime() + DOCUMENTATION_LIFETIME,
    });
```

Reply:
Yes, it's set to a future time so right now it's saying:
```
if (futureTime >= currentTime) {
  deleteCache()
}
```

Or:
> If the time to expire is after the current time, delete the cache.

----

We want it to keep the cache _until_ the time to expire has passed.

Reply:
omg, sorry...